### PR TITLE
[IOTDB-3191][IOTDB-3192] Refactoring sync module to use AbstractSyncInfo to manage pipe metadata

### PR DESCRIPTION
**Pull request by Cpaulyz**
### Overview

This pr modifies the sync architecture on the old standalone version to reduce the coupling of the individual component pieces. The old execution process will be replaced with a new one and adapted for distribution later. 

A comparison of the current architecture with the future architecture is illustrated below.
架构

### Common

* Extracting sync configuration to nodes-common
  * `sync_dir`
* Move some Utils and Constants to nodes-common
  * `SyncConstant.java`
  * `SyncPathUtil.java`
  * `SyncPipeUtil.java`

### Pipe

* Merge the former `ReceiverLog` and `SenderLogger` into a `SyncLogger`, with sync metadata managed by `AbstractSyncInfo`, which will be placed on Confignode in the future.
* Extract PipeInfo from Pipe. The original pipe has information and behavioural responsibilities, confignode only needs to manage the information in the Pipe.
* The architecture on the receiving end has been refactored.
  * The receive no longer maintains the pipe metadata, so some of the rpc has been removed.
  * Asynchronous loading have been changed to synchronous loading, there is no need to use PipeDataQueue at the receiver.
  * Note: *show pipe* is currently not supported on the receiver, it will be supported in the next PR.

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/sync/common/persistence/SyncLogger.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+/**
+ * SyncLogger is used to manage the persistent information in the sync module. Persistent
+ * information can be recovered on reboot via {@linkplain SyncLogAnalyzer}.
+ */
+public class SyncLogger {
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/sync/common/persistence/SyncLogger.java:37]:
```suggestion
public class SyncLogWriter {
```

File: server/src/main/java/org/apache/iotdb/db/sync/common/persistence/SyncLogger.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  // record pipe meta info
+  private BufferedWriter pipeWriter;
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/sync/common/persistence/SyncLogger.java:39]:
```suggestion
  private BufferedWriter pipeInfoWriter;
```

File: server/src/main/java/org/apache/iotdb/db/sync/common/persistence/SyncLogger.java (hunk 1 above)
Comment (Cpaulyz) [reply server/src/main/java/org/apache/iotdb/db/sync/common/persistence/SyncLogger.java]: done

File: server/src/main/java/org/apache/iotdb/db/sync/common/persistence/SyncLogger.java (hunk 2 above)
Comment (Cpaulyz) [reply server/src/main/java/org/apache/iotdb/db/sync/common/persistence/SyncLogger.java]: done

File: server/src/main/java/org/apache/iotdb/db/sync/common/persistence/SyncLogAnalyzer.java [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class SyncLogAnalyzer {
```
[the remark this answers is not in the document]
Comment (Cpaulyz) [reply server/src/main/java/org/apache/iotdb/db/sync/common/persistence/SyncLogAnalyzer.java]: done

File: server/src/main/java/org/apache/iotdb/db/sync/common/LocalSyncInfo.java [hunk 4]
```diff
@@ -16,10 +16,18 @@
...
  * under the License.
  */
-package org.apache.iotdb.db.exception.sync;
+package org.apache.iotdb.db.sync.common;
 
-public class PipeDataLoadBearableException extends PipeDataLoadException {
-  public PipeDataLoadBearableException(String message) {
-    super(message);
-  }
+public class LocalSyncInfo extends AbstractSyncInfo {
```
[the remark this answers is not in the document]
Comment (Cpaulyz) [reply server/src/main/java/org/apache/iotdb/db/sync/common/LocalSyncInfo.java]: Only SyncInfo is retained, not as an abstract class

File: server/src/main/java/org/apache/iotdb/db/sync/transport/server/TransportServiceImpl.java [hunk 5]
```diff
@@ -60,12 +56,12 @@
...
-import static org.apache.iotdb.db.sync.transport.conf.TransportConstant.RETRY_CODE;
-import static org.apache.iotdb.db.sync.transport.conf.TransportConstant.SUCCESS_CODE;
+import static org.apache.iotdb.commons.sync.SyncConstant.CONFLICT_CODE;
+import static org.apache.iotdb.commons.sync.SyncConstant.DATA_CHUNK_SIZE;
+import static org.apache.iotdb.commons.sync.SyncConstant.ERROR_CODE;
+import static org.apache.iotdb.commons.sync.SyncConstant.REBASE_CODE;
+import static org.apache.iotdb.commons.sync.SyncConstant.RETRY_CODE;
+import static org.apache.iotdb.commons.sync.SyncConstant.SUCCESS_CODE;
 
 public class TransportServiceImpl implements TransportService.Iface {
```
[the remark this answers is not in the document]
Comment (Cpaulyz) [reply server/src/main/java/org/apache/iotdb/db/sync/transport/server/TransportServiceImpl.java:66]: I will optimize it in the transport layer related issue later. 
(This has been noted in the issue: https://issues.apache.org/jira/browse/IOTDB-3950)

File: server/src/main/java/org/apache/iotdb/db/sync/sender/service/TransportHandler.java [hunk 6]
```diff
@@ -42,7 +37,6 @@
 import java.net.UnknownHostException;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Future;
-import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeUnit;
 
 public class TransportHandler {
```
[the remark this answers is not in the document]
Comment (Cpaulyz) [reply server/src/main/java/org/apache/iotdb/db/sync/sender/service/TransportHandler.java:42]: I will optimize it in the transport layer related issue later.
(This has been noted in the issue: https://issues.apache.org/jira/browse/IOTDB-3950)

File: server/src/main/java/org/apache/iotdb/db/sync/common/AbstractSyncInfo.java [hunk 7]
```diff
@@ -0,0 +1,301 @@
...
+  }
+
+  /**
+   * read the most important message about one pipe. ERROR > WARN > INFO.
+   *
+   * @param pipeName name of pipe
+   * @param createTime createTime of pipe
+   * @param consume if consume is true, recent messages will not be deleted. Otherwise, these
+   *     messages can be read next time.
+   * @return the most important message
```
[the remark this answers is not in the document]
Comment (Cpaulyz) [reply server/src/main/java/org/apache/iotdb/db/sync/common/AbstractSyncInfo.java]: Read the most important message about one pipe. ERROR > WARN > INFO.
(Message management related will be optimized in the sender related issue: https://issues.apache.org/jira/browse/IOTDB-3224)

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**